src/app/dataset-view/additional-components/data-component/load-more/load-more.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, EventEmitter, Input, Output } from "@angular/core";
+
+import AppValues from "../../../../common/app.values";
+
+@Component({
+    selector: "app-load-more",
+    templateUrl: "./load-more.component.html",
+    styleUrls: ["./load-more.component.sass"],
+})
+export class LoadMoreComponent {
+    public rowsNumber: number = AppValues.SQL_QUERY_LIMIT;
+    public isHolderVisible = false;
+
+    @Input() public isAllDataLoaded = false;
+
+    @Output() public loadMoreEmit = new EventEmitter<number>();
+
+    readonly ROWS_OPTIONS: number[] = [
+        AppValues.SQL_QUERY_LIMIT / 2,
+        AppValues.SQL_QUERY_LIMIT,
+        AppValues.SQL_QUERY_LIMIT * 2,
+    ];
+
+    public toggleOptions(): void {
+        this.isHolderVisible = !this.isHolderVisible;
+    }
+
+    public loadMore(): void {
```

Comment:
Missing tests for this handler. Very basic one, ensure that clicking on a button (not invoking this method, but clicking actually) results in @Output emitter notified.

Reply:
Added